### PR TITLE
[no-release-notes] doltcore/sqle/enginetest: Validating Enginetest Harness

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**
Initial implementation only targets the new format. One validation stage has been implemented: checking for dangling chunks.

### CODE REVIEW

File: go/libraries/doltcore/doltdb/durable/index.go [hunk 1]
```diff
@@ -147,6 +147,24 @@ type nomsIndex struct {
 
 var _ Index = nomsIndex{}
 
+func IterAllIndexes(
```
Comment (zachmu) [inline go/libraries/doltcore/doltdb/durable/index.go:150]: Isn't this a method on either Schema or IndexSet?

**Review by zachmu — APPROVED** [review]
LGTM

File: go/libraries/doltcore/doltdb/durable/index.go (hunk 1 above)
Comment (andrew-wm-arthur) [reply go/libraries/doltcore/doltdb/durable/index.go:150]: it is now!

**Comment by andrew-wm-arthur** [comment]
before:
```
% DOLT_DEFAULT_BIN_FORMAT="__DOLT_1__" go clean -testcache && go test ./...
ok      github.com/dolthub/dolt/go/libraries/doltcore/sqle/enginetest   40.727s

```
after:
```
% DOLT_DEFAULT_BIN_FORMAT="__DOLT_1__" go clean -testcache && go test ./...
ok      github.com/dolthub/dolt/go/libraries/doltcore/sqle/enginetest   43.598s
```

**Pull request merged**